### PR TITLE
fix(nuxt): preserve explicit `rel` attribute on internal link

### LINKED ISSUE #19299: nofollow is removed when internal links is set
### Environment

- Operating System: `Linux`
- Node Version:     `v18.14.0`
- Nuxt Version:     `3.2.2`
- Nitro Version:    `2.2.3`
- Package Manager:  `npm@9.5.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/github-c8trhm?file=pages/index.vue

### Describe the bug

When I set an internal link and rel="nofollow" to <NuxtLink />, rel="nofollow" is removed.

```html
<!-- code -->
<NuxtLink to="/a" rel="nofollow">to a</NuxtLink>

<!-- expected -->
<a href="/a" rel="nofollow">to a</a>

<!-- actual -->
<a href="/a" class="">to a</a>
```

https://github.com/nuxt/nuxt/blob/268cded/packages/nuxt/src/app/components/nuxt-link.ts#L213-L229
Looking at the code above, it seems that rel is excluded for internal links
However, sometimes you want to add rel="nofollow" even for internal links
For example, if an application needs to set a query to display a modal, but you don't want search engines to recognize it as a page.

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by igayamaguchi**
### 🔗 Linked issue

fix #19299

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Add `rel` attribute even if internal links is set.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by TheAlexLichter — APPROVED** [review]

File: packages/nuxt/src/app/components/nuxt-link.ts [hunk 1]
```diff
@@ -222,7 +222,8 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
               exactActiveClass: props.exactActiveClass || options.exactActiveClass,
               replace: props.replace,
               ariaCurrentValue: props.ariaCurrentValue,
-              custom: props.custom
+              custom: props.custom,
+              rel: props.rel
```
Comment (manniL) [inline packages/nuxt/src/app/components/nuxt-link.ts:226]: Should we use the `rel` from below (L241 and following)? 🤔

**Review by lihbr — APPROVED** [review]

File: packages/nuxt/src/app/components/nuxt-link.ts (hunk 1 above)
Comment (manniL) [reply packages/nuxt/src/app/components/nuxt-link.ts:226]: Good point 👍

**Comment by lihbr** [comment]
Awesome, thank you so much for your contribution!

**Comment by ydnar** [comment]
This change breaks (with heaps of warnings) if you use `<NuxtLink custom ...>` to specify a tag other than `<a>`.

```vue
	<NuxtLink custom v-slot="{ href, navigate }" :to="targetRoute">
		<component :is="tag" :href="href" @click="navigate" role="link" class="domain"
			><slot>{{ displayDomain(domain, status) }}</slot></component>
	</NuxtLink>
```

**Comment by nathanchase** [comment]
FYI: Something in this PR I believe broke NuxtLink (or perhaps https://github.com/nuxt/nuxt/pull/19309), and am now getting flooded with
[Vue warn]: Extraneous non-props attributes (rel) were passed to component but could not be automatically inherited because component renders fragment or text root nodes. anywhere that uses a custom NuxtLink template.

Example:

```
<NuxtLink
  v-slot="{ href }"
  :custom="true"
  :to="`/user/${id}`"
>
    <a :href="href">
      <img src="avatar.png">
    </a>
</NuxtLink>

```
The above doesn't error in 3.2.2, but does in 3.2.3.
Repro: https://stackblitz.com/edit/github-hmvahh?file=app.vue

**Comment by manniL** [comment]
@nathanchase see https://github.com/nuxt/nuxt/issues/19375 and https://github.com/nuxt/nuxt/pull/19379 which resolves that.

**Pull request merged**